sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1906,288 @@ def transform(self, X):
+                     bounds_error=False,
+                     fill_value=(min(self.references_),
+                                 max(self.references_)))
+            for quantiles_feat in self.quantiles_.T]
+
+        self.f_inverse_transform_ = [
+            interp1d(self.references_, quantiles_feat,
+                     bounds_error=False,
+                     fill_value=(min(quantiles_feat),
+                                 max(quantiles_feat)))
+            for quantiles_feat in self.quantiles_.T]
+
+    def _dense_fit(self, X):
+        """Compute percentiles for dense matrices.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            The data used to scale along the features axis.
+        """
+        rng = check_random_state(self.random_state)
+
+        # subsample the matrix X if necessary
+        if self.subsample < X.shape[0]:
+            subsample_idx = rng.choice(X.shape[0], self.subsample,
+                                       replace=False)
+        else:
+            subsample_idx = range(X.shape[0])
+
+        self.references_ = np.linspace(0, 1, self.n_quantiles,
+                                       endpoint=True)
+        self.quantiles_ = np.percentile(X[subsample_idx, :],
+                                        self.references_ * 100, axis=0)
+
+    def _sparse_fit(self, X):
+        """Compute percentiles for sparse matrices.
+
+        Parameters
+        ----------
+        X : sparse matrix, shape (n_samples, n_features)
+            The data used to scale along the features axis. The sparse matrix
+            needs to be semi-positive.
+        """
+        rng = check_random_state(self.random_state)
+
+        n_samples, n_feat = X.get_shape()
+        if self.subsample < n_samples:
+            subsample_idx = rng.choice(n_samples, self.subsample,
+                                       replace=False)
+            X_csr = X.tocsr()[subsample_idx]
+            X = X_csr.tocsc()
+
+        self.references_ = np.linspace(0, 1, self.n_quantiles,
+                                       endpoint=True)
+        # FIXME: it does not take into account the zero in the computation
+        self.quantiles_ = np.array([np.percentile(
+            X.data[X.indptr[feat]:X.indptr[feat + 1]], self.references_ * 100)
+                                    for feat in range(n_feat)]).T
+
+    def fit(self, X, y=None):
+        """Compute the quantiles used for normalizing.
+
+        Parameters
+        ----------
+        X : ndarray or sparse matrix, shape (n_samples, n_features)
+            The data used to scale along the features axis. If a sparse
+            matrix is provided, it will be converted into a sparse
+            ``csc_matrix``. Additionally, the sparse matrix needs to be
+            semi-positive.
+
+        Returns
+        -------
+        self : object
+            Returns self
+        """
+        X = self._validate_X(X)
```

Comment:
Just a niptick, is it necessary to create a specific function ? 
When there are few lines I prefer not create function ;).